### PR TITLE
chore: reduce scope of `#[allow(dead_code)]`

### PR DESCRIPTION
**Pull request by alamb**
I noticed the entire module allowed dead code and figured it would be good to reduce the scope

### CODE REVIEW

File: wal/src/lib.rs [hunk 1]
```diff
@@ -478,6 +476,7 @@ pub struct WriteSummary {
     /// Which segment file this entry was written to
     pub segment_id: SegmentId,
     /// Checksum for the compressed data written to segment
+    #[allow(dead_code)]
```
Comment (alamb) [inline wal/src/lib.rs:479]: This appears to be the only field not yet used -- @pauldix  I wonder how important is it to get the data checksummed?

**Review by crepererum — APPROVED** [review]

File: wal/src/lib.rs (hunk 1 above)
Comment (domodwyer) [reply wal/src/lib.rs:479]: It is checksummed, this is for the caller to use if needed (though clearly not) :+1:

**Review by domodwyer — APPROVED** [review]

File: wal/src/lib.rs (hunk 1 above)
Comment (alamb) [reply wal/src/lib.rs:481]: Do you think there is still value to the field? Maybe I could remove the field entirely and we could add it back if/when it is needed 🤔

Comment (domodwyer) [reply wal/src/lib.rs:481]: https://github.com/influxdata/influxdb_iox/pull/6840

**Pull request merged**